src/main/scala/org/allenai/deep_qa/data/BackgroundCorpusSearcher.scala

```diff
@@ -82,19 +82,39 @@ class LuceneBackgroundCorpusSearcher(params: JValue) extends BackgroundCorpusSea
   lazy val settings = Settings.builder().put("cluster.name", esClusterName).build()
   lazy val esClient = TransportClient.builder().settings(settings).build().addTransportAddress(address)
 
-  override def getBackground(query: String): Seq[String] = {
+  override def getBackground(query: Query): Seq[String] = {
+    // Make the bool query and the more-or-less canonical query string for the
+    // consolidation step (below)
+    val queryBuilder = QueryBuilders.boolQuery()
+    var queryString = ""
```

Comment:
In scala, avoid `var` wherever possible.  Using `val` removes potential sources of error and makes it easier to parallelize code.  So, you could write this as:

```scala
val queryString = query match {
  case stringQuery: StringQuery => {
    queryBuilder.should(...)
    stringQuery.query
  }
  case boostedQuery: BoostedQuery => {
    ...
    boostedQuery.boostedQueries.map(...)
  }
}